clang/lib/Basic/Targets/Sparc.h

```diff
@@ -151,7 +151,7 @@ class LLVM_LIBRARY_VISIBILITY SparcV8TargetInfo : public SparcTargetInfo {
 public:
   SparcV8TargetInfo(const llvm::Triple &Triple, const TargetOptions &Opts)
       : SparcTargetInfo(Triple, Opts) {
-    resetDataLayout("E-m:e-p:32:32-i64:64-f128:64-n32-S64");
+    resetDataLayout("E-m:e-p:32:32-i64:64-i128:128-f128:64-n32-S64");
```

Comment:
@koachan I realize I'm late to the party, but the original bug report in #102783 says that the bug is only for 64-bit SPARC. Was this change for 32-bit intentional?

Reply:
Ah, that makes sense. Thanks for the clarification.